aten/src/ATen/native/mkldnn/Conv.cpp

```diff
@@ -248,7 +242,8 @@ std::tuple<at::Tensor,at::Tensor,at::Tensor> mkldnn_convolution_backward(
     IntArrayRef padding_l, IntArrayRef padding_r, IntArrayRef stride,
     IntArrayRef dilation, int64_t groups, std::array<bool, 3> output_mask)
 {
-  Tensor grad_output = grad_output_t.contiguous();
+  const Tensor& grad_output = grad_output_t.is_mkldnn() ?
+      grad_output_t : grad_output_t.contiguous();
```

Comment:
`Tensor.contiguous()` here doesn't work for `mkldnn` tensors. There are also other places where the `grad`s are all returned as CPU tensors and autograd will raise an error about expecting `layout=Mkldnn`.

It seems the mkldnn backend is not actually tested properly with mkldnn tensors as inputs. Is that not the expected use case?